src/test/java/org/jenkinsci/plugins/gitclient/CredentialsTest.java

```diff
@@ -150,11 +147,6 @@ public void setUp() throws IOException, InterruptedException {
         listener = new hudson.util.LogTaskListener(logger, Level.ALL);
         listener.getLogger().println(LOGGING_STARTED);
         git = Git.with(listener, new hudson.EnvVars()).in(repo).using(gitImpl).getClient();
-        if (gitImpl.equals("git")) {
-            addExpectedLogSubstring("> git fetch ");
-            addExpectedLogSubstring("> git checkout -b master ");
-        }
-        addExpectedLogSubstring("Using reference repository: ");
```

Comment:
Please explain why you removed this assertion that confirms a reference repository is actually being used.

If you object to checking command line git output to confirm that a reference repository was actually used, then it needs to be replaced with a detailed check that the reference repository was used.  

A detailed check for use of a reference repository is more challenging, since it involves checking for the alternates file and confirming that one or more references are resolved through the alternates file rather than being copied into the objects directory.  Reference repositories are a crucial way that the git plugin reduces disc use and network data transfer.

Reply:
The cleanup consists of many small commits. This change here originates from dbcec24aca49797b45042dc69378d19067e4c60a. The commit message contains more description. This code was removed because the asseration `checkExpectedLogSubstring()` is not called anymore.